### DISCUSSION TITLE
2022/04/17/hexo-blog-3/

### DISCUSSION THREAD

**Discussion by giscus** [argvchs/giscus-comments#7] [Announcements]
# 2022/04/17/hexo-blog-3/

https://argvchs.netlify.app/2022/04/17/hexo-blog-3/

**Comment by match-fire** [comment]
666

**Comment by luhen123** [comment]
鼠标点击特效的代码放在哪里呢

**Reply by argvchs** [reply]
```
themes/particlex/source/js/fireworks.min.js
```

**Reply by yzr432** [reply]
js下没有fire这个文件是要自己创建吗，我创建了显示报错

**Comment by WindyDante** [comment]
<script src="<%- url_for(" /js/main.js") %>"></script>大佬，这行代码报错了Unterminated string literal，然后就加载不出来了,main.js直接跑不起来了，求助大佬

**Reply by WindyDante** [reply]
找到问题了，vscode自动保存给我路径加了个空格

**Reply by zhwuyuhehe** [reply]
我这里也有报错，我把代码修改如下就不再报错了。
`<script src="<%- url_for('/js/main.js') %>"></script>`
我觉得是因为两对双引号、导致了中间一个字符串成为了一个变量(/js/main.js取消了双引号，成为了非法变量)，但是不管我改不改这个双引号，都不影响最终的运行结果。
这里怎么写更恰当？

**Reply by argvchs** [reply]
@zhwuyuhehe EJS 会预处理被 `<%>` 括起来的内容，所以 `<%>` 的内外是互不干扰的

处理完之后这段代码就变成这样了

```html
<script src="/js/main.js"></script>
```

另外 VSCode 会自动将 EJS 错误识别为 HTML，不用管报错的内容

**Comment by nobuhiko233** [comment]
大佬 为什么我这里fireworks.min.js在网页中显示语法错误呢
![](https://cdnjson.com/images/2023/08/11/QQ20230811144838.png)

**Reply by argvchs** [reply]
教程里的 `fireworks.min.js` 是一个链接啊，把链接的内容填在这里

**Reply by nobuhiko233** [reply]
没学过三件套 问了愚蠢的问题 哈哈哈哈

**Comment by Huloveu** [comment]
你好大佬，我在添加鼠标点击格式以后，检查网页时有这个报错，鼠标点击格式没有加载，这是我的博客，请帮忙看一下https://huloveu.github.io/Blog/
GET https://huloveu.github.io/js/fireworks.min.js net::ERR_ABORTED 404

**Reply by zhwuyuhehe** [reply]
![image](https://github.com/argvchs/giscus-comments/assets/47172382/8b50adc7-89a3-49e1-a1e5-b87734006634)
### 目测你需要加上` /Blog `才能正常访问，现在你的JS文件是无法访问到的，加上/Blog之后就可以在浏览器中访问到。</br>
![image](https://github.com/argvchs/giscus-comments/assets/47172382/2350c607-8fd3-473e-832b-d0af24eb8d1f)
![image](https://github.com/argvchs/giscus-comments/assets/47172382/5b5c8ef6-0bd9-49bf-838e-ab048afef880)

**Reply by Huloveu** [reply]
谢谢哥们儿，我还想问一些其他小问题，可以请你加我qq吗
1742303395

**Reply by zhwuyuhehe** [reply]
不好意思，我并不是这个博客主题的开发者。您可以等开发者看到你的评论了，向他提问更多有关主题的问题。我也不一定能解决你的全部问题。

**Reply by Huloveu** [reply]
我的问题都解决了，再次感谢

**Comment by lxx897** [comment]
fireworks.min.js为什么没有这个文件

**Comment by MinecraftDream12345** [comment]
![](https://cdn.luogu.com.cn/upload/image_hosting/i6g1ahjp.png)
为什么我这边使用了tag功能之后报错啊（（

**Reply by MinecraftDream12345** [reply]
```
ERROR
TypeError: D:\Git\hexo\themes\particlex\layout\layout.ejs:40
    38|         <%- partial("menu") %>
    39|         <div id="main" :class="loading ? 'into-enter-from': 'into-enter-active'">
 >> 40|             <%- partial(type) %>
    41|             <%- partial("footer") %>
    42|         </div>
    43|         <% if (theme.preview.enable) { %>

D:\Git\hexo\themes\particlex\layout\tags.ejs:5
    3|     <div class="categories-tags">
    4|         <% let prev; %>
 >> 5|         <% site.tags.data.forEach(tag => { %>
    6|         <%
    7|             const colors = is_tag(tag.name)
    8|                 ? ["linear-gradient(120deg, #9abbf7 0%, #ffbbf4 100%)"]

site.tags.data.forEach is not a function
    at eval ("D:\\Git\\hexo\\themes\\particlex\\layout\\tags.ejs":18:23)
    at tags (D:\Git\hexo\node_modules\ejs\lib\ejs.js:703:17)
    at _View._compiledSync (D:\Git\hexo\node_modules\hexo\dist\theme\view.js:122:32)
    at _View.renderSync (D:\Git\hexo\node_modules\hexo\dist\theme\view.js:64:29)
    at Object.partial (D:\Git\hexo\node_modules\hexo\dist\plugins\helper\partial.js:27:17)
    at eval ("D:\\Git\\hexo\\themes\\particlex\\layout\\layout.ejs":54:17)
    at layout (D:\Git\hexo\node_modules\ejs\lib\ejs.js:703:17)
    at _View._compiled (D:\Git\hexo\node_modules\hexo\dist\theme\view.js:125:67)
    at _View.render (D:\Git\hexo\node_modules\hexo\dist\theme\view.js:50:21)
    at D:\Git\hexo\node_modules\hexo\dist\theme\view.js:57:31
    at tryCatcher (D:\Git\hexo\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (D:\Git\hexo\node_modules\bluebird\js\release\promise.js:547:31)
    at Promise._settlePromise (D:\Git\hexo\node_modules\bluebird\js\release\promise.js:604:18)
    at Promise._settlePromise0 (D:\Git\hexo\node_modules\bluebird\js\release\promise.js:649:10)
    at Promise._settlePromises (D:\Git\hexo\node_modules\bluebird\js\release\promise.js:729:18)
    at _drainQueueStep (D:\Git\hexo\node_modules\bluebird\js\release\async.js:93:12)
    at _drainQueue (D:\Git\hexo\node_modules\bluebird\js\release\async.js:86:9)
    at Async._drainQueues (D:\Git\hexo\node_modules\bluebird\js\release\async.js:102:5)
    at Async.drainQueues [as _onImmediate] (D:\Git\hexo\node_modules\bluebird\js\release\async.js:15:14)
    at process.processImmediate (node:internal/timers:476:21)
```

**Reply by jiaqishi-cmd** [reply]
你这个问题解决了吗，我也遇到了，categories和tags都出现了这个问题

**Reply by MinecraftDream12345** [reply]
把hexo的版本降下来，你应该是 7.0.0 的，改成低一点的版本比如 6.3.0 之类的

**Reply by jiaqishi-cmd** [reply]
非常感谢，问题解决了

**Comment by AeonT3rf** [comment]
问一下为什么我的markdown文件中的代码块不像博主的这样

**Comment by Revethere** [comment]
问一下
```ejs
<canvas
    id="fireworks"
    style="position: fixed; top: 0; left: 0; width: 100vw; height: 100vh; pointer-events: none; z-index: 32767"
></canvas>
<script src="/js/fireworks.min.js"></script>
```
是不是放在 ```layout.ejs``` 的 ```body``` 里面 ( ```fireworks.min.js``` 放在 ```/js``` 里面，与 ```main.js``` 同级 )
但是弄出来没报错但是无法交互 ( 没有任何效果 )

**Reply by zwdemz** [reply]
是的，我的有效果

**Comment by Dumb-Duck** [comment]
大佬，我想问下这个主题怎么改为中文的阿，或者可以中文翻译

**Reply by Revethere** [reply]
如果说的是导航栏的话直接在主题的 ```_config.yml``` 文件把 ```menu``` 里面的东西改成中文就行了

**Comment by Throne0826** [comment]
佬怎么更改代码块的语言啊 我用typora 写的md 里面自动备注了语言 但是转到网页上就是显示highlight语言

**Comment by Aircraft-carrier** [comment]
佬为什么我这个界面和你的还是有一些不一样呢？就是代码块没办法分行

**Reply by Tjaweiof-czh** [reply]
看看这个https://argvchs.github.io/2022/05/10/hexo-theme-particlex/

**Comment by xipishi0** [comment]
添加了自定义文件：fireworks.min.js
但需要将此文件挂在html上吗？怎么挂……

**Comment by BitterDuck123456** [comment]
https://i.imgur.com/Jg5U1FE.png
https://i.imgur.com/dnmAa6J.png
請問哪裡出錯了?

**Comment by youkemaola** [comment]
大佬，怎么下载文件啊，按上述代码，怎么都找不到相应文件，markdown里预览是有下载按钮的，但下载资源无论放在source根目录还是post内文章同一个文件夹里，编译后点击下载始终是找不到文件，比如报错“无法找到/post/MY/my.pdf”等等。我填的相对路径换了多种表达方式也都没效果...网上太少资料了，特求教

**Reply by youkemaola** [reply]
解决  这个评论咋删呢

**Comment by xingxixia** [comment]
想问一下标题怎么改，找了半天没找到。。。

**Reply by xingxixia** [reply]
我的标题一直是hexo......就主页旁边那个

**Reply by xingxixia** [reply]
好吧，突然反应过来为什么了，但是这里评论的好像不能撤回

**Comment by tangchengd** [comment]
你好 cursor.min.css的链接打不开

**Reply by argvchs** [reply]
https://static-argvchs.netlify.app/css/cursor.css
:O 这个应该是可以的

**Reply by tangchengd** [reply]
可以了 谢谢

**Reply by tangchengd** [reply]
还想问一下 点击文章的正文的背景图片怎么配置 因为我看有的用这个主题的博客 文章正文里有背景图片
